src/components/Contacts/ContactDetails/ContactDetailsTab/People/Items/EditPersonModal/PersonEmail/PersonEmail.tsx

```diff
@@ -59,33 +78,77 @@ export const PersonEmail: React.FC<PersonEmailProps> = ({ person }) => {
               </ContactPrimaryPersonSelectLabel>
               <Select
                 id="primary-email-label"
-                value={person.primaryEmailAddress?.email}
+                value={primaryEmail?.id}
+                onChange={(event) =>
+                  handleChangePrimary(event.target.value as string)
+                }
               >
-                {person.emailAddresses.nodes.map((emailAddress) => (
-                  <MenuItem key={emailAddress.id} value={emailAddress.email}>
-                    {emailAddress.email}
-                  </MenuItem>
-                ))}
+                {emailAddresses.map(
+                  (emailAddress) =>
+                    emailAddress.id && (
+                      <MenuItem key={emailAddress.id} value={emailAddress.id}>
+                        {emailAddress.email}
+                      </MenuItem>
+                    ),
+                )}
               </Select>
             </FormControl>
           </ModalSectionContainer>
-          {person.emailAddresses.nodes.map((emailAddress) => (
-            <>
-              <ModalSectionContainer>
-                <Grid container spacing={3}>
-                  <Grid item xs={6}>
-                    <ContactInputField value={emailAddress.email} fullWidth />
-                  </Grid>
-                  <Grid item xs={6}>
-                    <Select value={'Mobile'} fullWidth>
-                      <MenuItem value="Mobile">{t('Mobile')}</MenuItem>
-                    </Select>
-                  </Grid>
-                  <ModalSectionDeleteIcon />
-                </Grid>
-              </ModalSectionContainer>
-            </>
-          ))}
+          <FieldArray
+            name="emailAddresses"
+            render={() => (
+              <>
+                {emailAddresses?.map((emailAddress, index) => (
+                  <>
+                    <ModalSectionContainer key={index}>
+                      <Grid container spacing={3}>
+                        <Grid item xs={6}>
+                          <ContactInputField
+                            value={emailAddress.email}
+                            onChange={(event) =>
+                              setFieldValue(
+                                `emailAddresses.${index}.email`,
+                                event.target.value,
+                              )
+                            }
+                            error={getIn(errors, `emailAddresses.${index}`)}
+                            helperText={
+                              getIn(errors, `emailAddresses.${index}`) &&
+                              t(getIn(errors, `emailAddresses.${index}`).email)
```

Comment:
`t` doesn't work with dynamic props. Something static needs to be extracted. You can translate custom things when you pass them to yup above. Not sure what to do if yup has default values.

Reply:
Oh, ignore my other comments then. I try to figure something out then.